api/v1/helpers/verbs/doPatch.js

```diff
@@ -29,21 +30,35 @@ const logAPI = require('../../../../utils/loggingUtil').logAPI;
  *  resource type to patch.
  */
 function doPatch(req, res, next, props) {
+  let modelInst;
   const requestBody = req.swagger.params.queryBody.value;
   u.findByKey(props, req.swagger.params)
   .then((o) => {
+    modelInst = o;
+    return featureToggles.isFeatureEnabled('enforceWritePermission') ?
+                u.isWritable(req, o) : true;
+  })
+  .then((ok) => {
+    if (!ok) {
+      return u.forbidden(next, props.modelName);
+    }
+
     // To avoid timeouts when patching samples; force the update, even if
     // the value has not changed. Adding this to the "before update hook" does
     // give the needed effect; so adding it here!!!.
     if (props.modelName === 'Sample') {
-      o.changed('value', true);
+      modelInst.changed('value', true);
     }
 
-    u.patchJsonArrayFields(o, requestBody, props);
-    u.patchArrayFields(o, requestBody, props);
-    return o.update(requestBody);
+    u.patchJsonArrayFields(modelInst, requestBody, props);
+    u.patchArrayFields(modelInst, requestBody, props);
+    return modelInst.update(requestBody);
   })
   .then((retVal) => {
+    if (!retVal) {
+      return null;
```

Comment:
Just curious, in what case does the execution control reaches here (when retVal is null)